R/rampReactionQueries.R

```diff
@@ -841,19 +927,30 @@ runReactionClassTest <- function( analytes,
                                 "mets_pval" = ec_level_2_stats$pvals_mets,
                                 "mets_oddsratio" = as.numeric(ec_level_2_stats$oddsratio_mets))
 
-      out <- data.frame(rbind(ec_level_1_stats, ec_level_2_stats))
+      #Calculate adjusted pvals independently for each EC level
+      adjusted_stats <- function(reactionClass_stats)
```

Comment:
adjusted_stats is defined twice with almost the same function body
probably you can put a parameter for prot_pval or mets_pval, or do something else to combine them so you only have to define it once

Reply:
I consolidated both the runFisherReaction() and adjusted_stats() functions below. I also brought them out as helper functions instead of keeping them nested in.